packages/lib/pod/src/podUtil.ts

```diff
@@ -175,17 +178,42 @@ export function requireType(
   value: unknown,
   typeName: string
 ): void {
-  if (typeof value !== typeName) {
-    throw new TypeError(
-      `Invalid value for entry ${nameForErrorMessages}.  Expected type ${typeName}.`
-    );
+  switch (typeName) {
+    case "object":
+      if (typeof value !== "object" || Array.isArray(value) || value === null) {
+        throw new TypeError(
+          `Invalid value for entry ${nameForErrorMessages}.  \
+          Expected a non-array non-null object.`
+        );
+      }
+      break;
+    case "array":
+      if (typeof value !== "object" || !Array.isArray(value)) {
+        throw new TypeError(
+          `Invalid value for entry ${nameForErrorMessages}.  \
+          Expected an array.`
+        );
+      }
+      break;
+    default:
+      if (typeof value !== typeName) {
+        throw new TypeError(
+          `Invalid value for entry ${nameForErrorMessages}.  \
+          Expected type ${typeName}.`
+        );
+      }
+      break;
   }
 }
 
 /**
  * Checks that `value` has the run-time type given by `typeName`.  Compile-time
  * type of input/output is limited to expected POD value types.
  *
+ * Works for any runtime JavaScript type, but two values have special meaning.
```

Comment:
I guess this should agree with the comment before `requireType` so maybe just refer to that?